server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/rfc/api/routes/JMAPApiRoutes.scala

```diff
@@ -1,33 +1,50 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
 package org.apache.james.jmap.rfc.api.routes
 
+import io.netty.handler.codec.http.HttpResponseStatus
 import org.apache.james.jmap.rfc.api.method.EchoMethod
-import org.apache.james.jmap.rfc.model.{RequestObject, ResponseObject}
-import org.slf4j.{Logger, LoggerFactory}
 import reactor.core.publisher.Mono
+import reactor.netty.http.server.{HttpServerRequest, HttpServerResponse}
 
 object JMAPApiRoutes {
-  private val LOGGER = LoggerFactory.getLogger(JMAPApiRoutes.getClass)
   private val ECHO_METHOD = new EchoMethod();
+  private val METHOD_NAME_PARAMETER = "method-name"
 }
 
 class JMAPApiRoutes {
-  val logger: Logger = JMAPApiRoutes.LOGGER
-  val echoMethod: EchoMethod = JMAPApiRoutes.ECHO_METHOD
-
-  def post(httpRequest: HttpServerRequest, httpServerResponse: HttpServerResponse): Mono[Void] = {
-    httpRequest
-      .receive()
-      .asInputStream()
-      .flatMap(inputStream => {
-        echoMethod.toRequestObject(inputStream)
-      })
-      .flatMap((requestObject: RequestObject) => {
-        echoMethod.toResponseObject(requestObject)
-          .map((responseObject: ResponseObject) => {
-            httpServerResponse.status(HttpResponseStatus.OK).sendObject(responseObject)
-          })
-      })
+  def post(httpServerRequest: HttpServerRequest, httpServerResponse: HttpServerResponse): Mono[Void] = {
+    Mono.just(httpServerRequest)
```

Comment:
You don't need the Mono.just and the following flatMap. No ?
`this.process(httpRequest, httpServerResponse)` should be enough